merkle_zeppelin/trees/merkle.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, Callable
+
+from Crypto.Hash import keccak
+from eth_abi import encode
+
+from .binary import BinaryTree
+from .exceptions import MerkleTreeValidationFailed, ValueNotFoundInTree
+
+Leaf = tuple[Any, ...]
+
+
+def keccak256(v: bytes) -> bytes:
+    return keccak.new(data=v, digest_bits=256).digest()
+
+
+class MerkleTree(BinaryTree[bytes]):
+    def __init__(
+        self,
+        leaves: list[Leaf],
+        types: list[str],
+        hashing_function: Callable[[bytes], bytes] = None,
```

Comment:
Just wanted to mention I usually like using [strict optional](https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_optional) in mypy. It helps catch places where we might forget to check for 'None', and avoids those 'NoneType' attribute errors.
```suggestion
        hashing_function: Callable[[bytes], bytes] | None = None,
```

Also, worth mentioning, there's [implicit_optional](https://mypy.readthedocs.io/en/stble/config_file.html#confval-implicit_optional). It's cool because it automatically makes arguments with None defaults optional and that's why probably there is no any mypy warning right now for this line.. But, it's only for arguments, not for other stuff. I usually prefer to write out Optional anyway – just seems clearer to me. But hey, it's up to you!

Reply:
so I used explicit `Optional`, but thanks for mentioning!